### PR TITLE
Validate ITRF structure instead of maintaining a support list

### LINKED ISSUE #7: Validator for ITRF values
Instead of using a hard-coded list of ITRF values, write a validator. See https://github.com/nsidc/iceflow/blob/pandera-spike/src/iceflow/itrf/__init__.py#L5-L10.

A hard-coded list is fine for now, but in the future there could be a new realization that e.g., ICESat2 uses that users will want to align their data to. As currently written, this would not be possible without a change to the code.

### PR DESCRIPTION
**Pull request by trey-stafford**
Also, make it so that pandera actually validates constraints! Using the covariant type approach was resulting in validations not running.

Resolves #7

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -110,6 +110,7 @@ ignore = [
   "EM101",    # Exceptions should not use string literals
   "SIM108",   # Use ternary operator. Sometimes this is less readable.
   "T201",     # Remove 'print' statements. Sometimes these are handy.
+  "SIM105",   # Use contextlib.suppress instead of try-except-pass. The recommended approach is slower and less flexible.
```
Comment (mfisher87) [inline pyproject.toml:113]: I dislike this rule too. We have an open PR on icepyx where we're discussing this! https://github.com/icesat2py/icepyx/pull/572

File: src/iceflow/data/atm1b.py [hunk 2]
```diff
@@ -312,16 +311,14 @@ def extract_itrf(filepath: Path) -> str:
...
-            err = f"Unrecognized ATM1B ITRF {itrf}"
-            raise RuntimeError(err) from e
+    # Try to normalize based on known ITRF strings in the ATM1B data:
+    try:
+        itrf = {
+            "ITRF05": "ITRF2005",
+            "ITRF08": "ITRF2008",
+        }[itrf]
+    except KeyError:
+        pass
```
Comment (mfisher87) [inline src/iceflow/data/atm1b.py:321]: Can we move this into a Pydantic validator?

File: tests/unit/test_itrf.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from __future__ import annotations
+
+from iceflow.itrf import check_itrf
+
+
+def test_check_itrf():
+    assert check_itrf("Not an ITRF string") is False
+    assert check_itrf("ITRF") is False
```
Comment (mfisher87) [inline tests/unit/test_itrf.py:8]: No preference, just thinking out loud. What about `assert not check_itrf("bad itrf")` and `assert check_itrf("good itrf")`?

**Review by mfisher87 — APPROVED** [review]

File: src/iceflow/data/atm1b.py (hunk 2 above)
Comment (trey-stafford) [reply src/iceflow/data/atm1b.py:321]: Perhaps in the future, I'll make a comment. I expect to do some additional refactoring around ITRF normalization to support other dataproducts soon.

File: tests/unit/test_itrf.py (hunk 3 above)
Comment (trey-stafford) [reply tests/unit/test_itrf.py]: ad67cae

**Pull request merged**